index.bs

```diff
@@ -4324,8 +4326,8 @@ partial interface MLGraphBuilder {
 
 <div>
     **Arguments:**
-        - *a*: an {{MLOperand}}. The first N-dimensional input tensor.
-        - *b*: an {{MLOperand}}. The second N-dimensional input tensor.
+        - *a*: an {{MLOperand}}. The first input tensor which is at least 2-D.
+        - *b*: an {{MLOperand}}. The second input tensor which is at least 2-D.
 
```

Comment:
~~Should we also mention that the two input tensors must have the same rank~~ Oh never mind, we simplified GEMM *some*, but we are still supporting broadcasting for it (part of me wanted to even have this higher-level broadcasting policy broadcasting be resolved by the caller, such that the lower level WebNN API only accepts consistent rank tensors here which is more easily testable/spec'able, but that probably borders on the side of caller inconvenience).

Reply:
[XLA has a stance on this](https://openxla.org/xla/broadcasting#principles) which I find very appealing:

> The XLA language is as strict and explicit as possible, avoiding implicit "magical" features. Such features might make some computations slightly easier to define, but at the cost of more assumptions baked into user code that will be difficult to change in the long term. If necessary, implicit magical features can be added in client-level wrappers.

If there's a tradeoff to be made between:
- caller inconvenience, and
- spec maintainability/testability, while removing ambiguity

... I think we should strive for the latter in WebNN :)

Reply:
Yeah, generally we've been moving more that direction. For example, `reshape` no longer takes a special null value as a wildcard to compute based on the tensor element count and the remaining elements. `matmul` (as you see here) lost its additional if/else if/else ~~1-D~~ 1D logic, with some cost to the caller (reshape calls). As someone who's implemented a number of specs (Unicode bidi, Unicode line breaking...) I definitely value spec clarity and clear conformance, and I appreciate every operator definition having clear pseudocode or a decomposition of equivalent operators. There *may* be some implementation-defined aspects for certain data types near the edges (depending on device capability), like saturation vs infinity, rounding of subnormals, and NaN representation, but generally yeah, we all want to lessen that.